[CS-4482] Disabling confirm button while loading data

oidoug opened the pull request:
### Description

PR to improve data loading on RegisterMerchantDisplay transaction sheet.

- Changes `usePrepaidCard` hook to use `useGetSafesDataQuery` that has probably already cached safes, instead of `useGetPrepaidCardsQuery` that would be a new query;
- Adds isLoading for `PayThisAmountSection` on `RegisterMerchantDisplay` too;
- Adds skeleton to Profile header info;
- Passes `disabledConfirmButton` on Register to disable pay section until loading done.

Note:

As it is now, we can't determine for certain when individual sections finish loading, they don't have a way to report back. It's imperceptible, but confirm button is still being enabled before all is loaded. Fixing this is honestly not worth it since loading is shown in all sections now and it's usually fast.

- [x] Completes #(CS-4482)

### Checklist

- [x] Tested on a small device
- [x] Tested on iOS
- [x] Tested on Android

### Screenshots

https://user-images.githubusercontent.com/129619/186911011-41d25b37-8d52-4570-ac8a-6aab37bdf882.mp4

oidoug committed:
Includes data loading also as loading flag to confirm tx button

danibonilha commented on cardstack/src/components/TransactionConfirmationSheet/TransactionConfirmationSheet.tsx:
```diff
@@ -146,7 +147,7 @@ const SheetFooter = ({
       </Button>
       <Button
-        loading={onConfirmLoading}
+        loading={onConfirmLoading || loading}
```
What are we trying to achieve here ? I thought we wanted to disabled the button, while the state was loading, so this prop should go on disabled

oidoug replied:
I see, we use loading in the button only after the action was taken, is that it? I never consider that, that's why I tend to use loading in the button instead of disable.

The goal is to avoid confirming before data was all presented to the user to decide.

danibonilha replied:
Yes, that feels more intuitive, loading on btn is the result of the action

paulinhapenedo commented on cardstack/src/components/TransactionConfirmationSheet/TransactionConfirmationSheet.tsx:
```diff
@@ -146,7 +147,7 @@ const SheetFooter = ({
       </Button>
       <Button
-        loading={onConfirmLoading}
+        loading={onConfirmLoading || loading}
```
Shouldn't the `loading` be passed to the `disabled` prop instead? This will solve the loading problem of the first screenshot on the ticket. 

About the second loading, @danibonilha and I were discussing about it and she gave an advice to change the `PrepaidCardTransactionSection` query from `usePrepaidCard` to the `useGetSafesDataQuery` like we have on `useChoosePrepaidCard`. Since safes info are cached, we may not even see the second loading happening (or it will be a bit faster, at least).

oidoug replied:
Nice, that makes sense, I'll try that.

oidoug committed:
Changes usePrepaidCard hook to use useGetSafesDataQuery hook that has probably already cached safes, instead of useGetPrepaidCardsQuery.
Adds isLoading for Paythisamountsection on RegisterMerchantDisplay too.
Adds skeleton to Profile header info.

oidoug commented:
@paulinhapenedo @danibonilha , I updated the PR and the description, please take a look at it again.
Thanks for the comments, they helped me find a path forward 😊 👍

paulinhapenedo reviewed (approved):
It's looking way better! 👏
